### ISSUE TITLE
Enforcing old C++11 standard makes harfbuzz incompatible with ICU

### ISSUE THREAD

**Issue by intractabilis** [harfbuzz/harfbuzz#4737]
https://github.com/harfbuzz/harfbuzz/blob/ec437ccd72f7c76155cacb3b6626117d6e85ee98/CMakeLists.txt#L531
This line downgrades the compiler capabilities to the ancient C++11. As a result, the compilation fails because other libraries need more modern than C++11 features.
```
FAILED: src/libharfbuzz-icu.so.0.60850.0.p/hb-icu.cc.o
g++ -Isrc/libharfbuzz-icu.so.0.60850.0.p -Isrc -I../harfbuzz/src -I. -I../harfbuzz -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++11 -fno-exceptions -O3 -fno-exceptions -fno-rtti -fno-threadsafe-statics -fvisibility-inlines-hidden -DHAVE_CONFIG_H -mtune=native -march=native -fPIC -Wno-non-virtual-dtor -MD -MQ src/libharfbuzz-icu.so.0.60850.0.p/hb-icu.cc.o -MF src/libharfbuzz-icu.so.0.60850.0.p/hb-icu.cc.o.d -o src/libharfbuzz-icu.so.0.60850.0.p/hb-icu.cc.o -c ../harfbuzz/src/hb-icu.cc
In file included from /usr/include/unicode/uset.h:36,
                 from /usr/include/unicode/unorm2.h:35,
                 from ../harfbuzz/src/hb-icu.cc:39:
/usr/include/unicode/localpointer.h:561:26: error: parameter declared ‘auto’
  561 | template <typename Type, auto closeFunction>
      |                          ^~~~
```

**Comment by alerque** [comment]
Yes, this has been reported in #4671, and also fixed already in 93d58f8, but only for the Meson build option. It loos like the CMake build will need a similar fix, but I think the CMake status is very much a community contribution. PRs welcome to get this fixed up there.

**Comment by intractabilis** [comment]
I had a brain fart. Yes, the change was needed in the Meson build file. It's great that it is already fixed. Thank you! Closing.
